### DISCUSSION TITLE
DateField onChange get day, month and year as one

### DISCUSSION THREAD

**Discussion by marksmall** [govuk-react/govuk-react#1295] [Q&A] [answered]
I'm looking to use the `DateField` in a form. I can see there is the `input` prop, that has an `onChange`, if I console.log the event, I get an object with only the current input either, day, month or year and the others are undefined, even if I have already typed values into them. I also know there is the `inputs` prop, that has day/month/year props, in which I could update some local state and stitch it all together when the form is submitted, but that seems really weird and wasteful, leading me to think I've missed something.

What I really want to do is have the user type the day, month and year and get the all values, as an object, string, I don't really care, so I can create an [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) datetime string from them to be used in a form submission, so what am I missing?

**Comment by penx** [comment, marked as answer]
From memory, if the form inputs are controlled (have values and onChange) then the onChange event should contain day, month and year. 

I think you raise a good point that this may not behave that way for uncontrolled inputs. 

I’d accept a PR to address that if you want to fix it. 

Alternatively I would recommend either:

- if you want a simple solution to use the fate field in uncontrolled mode, get the form data from the form element as described at https://developer.mozilla.org/en-US/docs/Web/API/FormData/Using_FormData_Objects#retrieving_a_formdata_object_from_an_html_form
- alternatively check out the examples folder which has examples using the date field with Formik, Final Form and react hook form

**Reply by marksmall** [reply]
I wasn't aware of the example app, so that's great, I can use the react-hook-form as a guide, thanks.

**Comment by marksmall** [comment]
Sorry to labour this, I have looked at the example app, from what I see it can only be used with cypress, but if I'm wrong on that, please say how to start the app. 

It looks like the it covers most situations, but I don't see it managing pre-populating a form. The `react-hook-form` example has a custom component wrapping the `GovUK.DateField` and it attempts to set the value state to that from the `register` function, except register never has the value, so that cannot work, meaning even if you have date values, they cannot be set on the component. Of course that is only an issue in the example, but it is a very common use-case and should be catered for.

I am working on this in my local app, if I get further I can report back here and maybe, time permitting, raise a PR.

**Reply by penx** [reply]
The example app can be viewed in storybook https://govuk-react.github.io/govuk-react/?path=/docs/example-application--docs
